samples/containers/app-raven/src/main/java/com/squareup/sample/ravenapp/RavenActivity.kt

```diff
@@ -41,3 +47,20 @@ class RavenActivity : AppCompatActivity() {
     }
   }
 }
+
+class RavenModel(savedState: SavedStateHandle) : ViewModel() {
+  private val _running = MutableStateFlow(true)
+  val running: StateFlow<Boolean> = _running
```

Comment:
This could just be a `CompletableJob`/`Job` which expresses the contract a little more accurately I think. 

Then the previous code would be 
```
running.join()
finish ()
```

Reply:
I like it, though if feels a little funny to expose an object that leaks its `complete()` method like that. Doing it anyway.

Reply:
What if we gave the model its own `suspend fun` for this? 

```kotlin
suspend fun waitForExit()
```

Reply:
From another sample. This has to be two separate `launch` calls, right? Otherwise the `collect` will never terminate and we'll never finish.

```kotlin
    lifecycleScope.launch {
      model.renderings.collect { Timber.d("rendering: %s", it) }
    }
    lifecycleScope.launch {
      model.waitForExit()
      finish()
    }
```